### PR TITLE
[TEST] Added Mock tests for joinRoom.js backend socket

### PR DESCRIPTION
**Pull request by David-Xia0**
Added Unit Tests test for sockets/joinRoom.js

Mocks created:
__mocks__/Games.js

Added Socket.io-client library to backend for testing purposes

Fixed a bug in socket/joinRoom.js

Ref: https://medium.com/@tozwierz/testing-socket-io-with-jest-on-backend-node-js-f71f7ec7010f
https://jestjs.io/docs/mock-functions#using-a-mock-function

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
     "dotenv": "^8.2.0",
     "express": "^4.17.1",
     "mongoose": "^5.11.19",
-    "socket.io": "^3.1.2"
+    "socket.io": "^3.1.2",
+    "socket.io-client": "^4.0.0"
```
Comment (Tianrens) [inline backend/package.json:22]: put it in dev dependency?

**Review by Tianrens — APPROVED** [review]

File: backend/package.json (hunk 1 above)
Comment (David-Xia0) [reply backend/package.json:22]: :(

Comment (David-Xia0) [reply backend/package.json]: Fixed

**Review by Tianrens — APPROVED** [review]
LGTM!

File: backend/package.json (hunk 1 above)
Comment (Tianrens) [reply backend/package.json]: Nice

**Review by prabhnandan — APPROVED** [review]

**Review by afei088 — APPROVED** [review]

**Review by Tianrens — APPROVED** [review]
LGTM

**Review by afei088 — APPROVED** [review]

**Comment by afei088** [comment]
beautiful work!

**Comment by prabhnandan** [comment]
LGTM

**Pull request merged**